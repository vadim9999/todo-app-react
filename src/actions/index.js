import { ADD_TASK, FOUND_WORD } from "../constants/action-types";


// export function addArticle(payload){
//     return { type: ADD_ARTICLE, payload}
// }


export function addTask(payload){
    console.log("Call Function in actions");
    console.log(payload);
    
    return { type: ADD_TASK, payload}
}

export function addFoundWord(payload){
    console.log("Found word");
    console.log(payload);
    
    return { type: FOUND_WORD, payload}
}

export function getData(){
    return fetch("https://jsonplaceholder.typicode.com/posts").then(
        response => response.json())
        .then(json => {
            return {
                type:"Data_LOADED",
                payload:json
            }
        })
    
}