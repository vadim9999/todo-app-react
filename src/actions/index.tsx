import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

// export function addArticle(payload){
//     return { type: ADD_ARTICLE, payload}
// }

export function addTask(payload: any) {
  return { type: ADD_TASK, payload };
}

export function addFoundWord(payload: any) {
  return { type: FOUND_WORD, payload };
}

export const getTasks = (payload: any) => ({ type: 'GET_TASKS', payload });

export const deleteTaskById = (payload: any) => ({
  type: 'DELETE_TASK',
  payload
});

export const updateTaskById = (payload: any) => ({
  type: 'UPDATE_TASK',
  payload
});
// export function getData() {
//     return function(dispatch) {
//       return fetch("https://jsonplaceholder.typicode.com/posts")
//         .then(response => response.json())
//         .then(json => {
//             console.log("data");
//             console.log(json);

//           dispatch({ type: "DATA_LOADED", payload: json });
//         });
//     };
//   }

export const sortTasksByGrowthDate = (payload: any) => ({
  type: 'SORT_TASKS_BY_GROWTH_DATE',
  payload
});

export const sortTasksByDecreaseDate = (payload: any) => ({
  type: 'SORT_TASKS_BY_DECREASE_DATE',
  payload
});

export const addCurrentPage = (payload: any) => ({
  type: 'ADD_CURRENT_PAGE',
  payload
});

export const addSelectedRowKeys = (payload:any) =>({
  type:'ADD_SELECTED_ROW_KEYS',
  payload
})
export const addCurrentSelectedRowKeys = (payload: any) => ({
  type: 'ADD_CURRENT_SELECTED_ROW_KEYS',
  payload
});
// export function saveTask(){
//   return { type: "SAVE_TASK"}
// }
