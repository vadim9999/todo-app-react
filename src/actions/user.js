import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

export const addUser = (payload) => ({
  type: 'ADD_USER',
  payload,
});

export const authenticate = (payload) => ({
  type: 'AUTHENTICATE', payload,
});

export const authorizate = (payload) => ({
  type: 'AUTHORIZATE', payload,
});
