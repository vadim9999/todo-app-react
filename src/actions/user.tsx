// import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

export const addUser = (payload:any) => ({
  type: 'ADD_USER',
  payload,
});

export const authenticate = (payload:any) => ({
  type: 'AUTHENTICATE', payload,
});

export const authorizate = (payload:any) => ({
  type: 'AUTHORIZATE', payload,
});
