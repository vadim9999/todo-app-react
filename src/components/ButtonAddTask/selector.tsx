
export const getPagination = (tasks: object[]) => {
  let counter = 0;

  if(tasks.length === 0){
    return 1
  } else {
    tasks.forEach((elem, index) => {
      console.log(index);
        if (index % 10 === 0) {
          counter++;
          // return(<Item onClick= {this.onClick} key = {elem["_id"]} id={counter++}>{counter}</Item>)
        }
      
    });
  }
  
  return counter;
};

