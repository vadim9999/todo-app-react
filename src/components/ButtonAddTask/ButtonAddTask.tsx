import React from 'react'

import {Button} from 'antd'
import moment from 'moment'
import {connect} from 'react-redux'
import {addTask, addCurrentPage} from '../../actions'

import {TasksTypes} from '../Interfaces'
import {getPagination} from './selector'
interface ButtonAddTaskProps {
  tasks: TasksTypes[];
  user_id: string;

  addTask: any;
  addCurrentPage: any;

}
const mapStateToProps = (state:any) =>{
  return {
    user_id: state.user._id,
    tasks: state.tasks
  }
}

const mapDispatchToProps = (dispatch:any) =>{
  return {
    addTask : (task:TasksTypes) => dispatch(addTask(task)),
    addCurrentPage: (page:number) => dispatch(addCurrentPage(page))
  }
}

export class ConnectedButtonAddTask extends React.Component<ButtonAddTaskProps,{}>{

  handleAdd = (e: any) => {
    e.preventDefault();
    const { tasks } = this.props;

    const date = moment().toISOString();

    this.props.addTask({
      login_id: this.props.user_id,
      completed: false,
      name: 'New task',
      date
    });

    const page = getPagination(tasks);
    this.props.addCurrentPage(page);
  };

  render(){
    return (
      <Button onClick={this.handleAdd} type="primary">
            Add a row
        </Button>
    );
  }
    
  
}

const ButtonAddTask = connect(mapStateToProps,mapDispatchToProps)(ConnectedButtonAddTask)
export default ButtonAddTask