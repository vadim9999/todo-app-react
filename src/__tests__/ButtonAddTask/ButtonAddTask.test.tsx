import React from 'react'

import Enzyme, {mount} from 'enzyme'
import configureStore from 'redux-mock-store'
import Adapter from 'enzyme-adapter-react-16'
import {Provider} from 'react-redux'
import ButtonAddTask,{ConnectedButtonAddTask} from '../../components/ButtonAddTask/ButtonAddTask'
import {addTask, addCurrentPage} from '../../actions'
import renderer from 'react-test-renderer'
import {Button} from 'antd'

Enzyme.configure({adapter: new Adapter()})

describe('Testing ButtonAddTask', () =>{
  const initialState = {
    user: {
      _id:'1234'
    },
    tasks: [],
    currentPage: 1,
    selectedRowKeys: [],
    currentSelectedRowKeys: []

  }

  const mockStore = configureStore()

  let store:any, wrapper:any;

  beforeEach(() =>{
    store = mockStore(initialState)

    wrapper = mount(
      <Provider store={store}>
        <ButtonAddTask />
      </Provider>
    )
  }) 

  it('check on render button', () =>{
    expect(wrapper.find(ButtonAddTask).length).toEqual(1)
  })

  it('check disptaching addTask',() =>{
    let action 
    store.dispatch(addTask({
      login_id: '134',
      completed: false,
      name: 'New task',
      date: '134:33'
    }))

    action = store.getActions()
    expect(action[0].type).toBe('ADD_TASK')
  })

  it('check dispatching addCurrentPage', ()=>{
    let action

    store.dispatch(addCurrentPage(1))

    action = store.getActions()

    expect(action[0].type).toBe('ADD_CURRENT_PAGE')
  })

  it('check props matches with initState', ()=>{
    expect(wrapper.find(ConnectedButtonAddTask).prop('tasks')).toEqual(initialState.tasks)
    expect(wrapper.find(ConnectedButtonAddTask).prop('user_id')).toEqual(initialState.user._id)
  })

  it('check on click Button',()=>{
    const button = wrapper.find(Button)
   const result =  button.prop('onClick')({preventDefault:jest.fn()})
    expect(result).toBe(undefined)
  })

  describe('check snapshot',()=>{
    it('snapshot',()=>{
      const renderedValue = renderer.create(<ConnectedButtonAddTask {...initialState} />).toJSON()
      expect(renderedValue).toMatchSnapshot()
    })
  })

})