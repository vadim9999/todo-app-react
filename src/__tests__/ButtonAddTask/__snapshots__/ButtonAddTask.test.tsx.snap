// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonAddTask check snapshot snapshot 1`] = `
<button
  className="ant-btn ant-btn-primary"
  onClick={[Function]}
  type="button"
>
  <span>
    Add a row
  </span>
</button>
`;
